mvg/analysis_classes.py

```diff
@@ -100,21 +133,30 @@ def _add_datetime_df(self, dframe, timecolumn):
         """
 
         # Check if there is info for conversion
-        if self.t_zone is not None:
+        if self._t_zone is not None:
             # EPOCH to datetime
             dframe = dframe.assign(
-                datetime=pd.to_datetime(dframe[timecolumn], unit=self.t_unit, utc=True)
+                datetime=pd.to_datetime(dframe[timecolumn], unit=self._t_unit, utc=True)
             )
             # Set timezone
             dframe = dframe.assign(
-                datetime=(dframe["datetime"].dt.tz_convert(self.t_zone))
+                datetime=(dframe["datetime"].dt.tz_convert(self._t_zone))
             )
             # Mark datetime as availble
             self.time_column = "datetime"
 
         return dframe
```

Comment:
I think this will apply the changes inplace for the input dataframe `dframe`. Better to create a new variable and return that

Reply:
Thanks, I simplified that code